book/src/rfcs/benchmarking.md

```diff
@@ -0,0 +1,266 @@
+
+Next the classes of benchmarks to consider, they are different combinations of
+metrics, layers and consumers.
+
+| benchmarks               | repository    | metric   | for external use | for internal use | regression test | noise sensitive  |
+| ------------------------ | ------------- | -------- | ---------------- | ---------------- | --------------- | ---------------- |
+| kimchi time              | proof-systems | time     | yes              | yes              | yes             | yes              |
+| kimchi time flamegraph   | proof-systems | time     | no               | yes              | no              | no               |
+| kimchi instructions      | proof-systems | time     | no               | yes              | yes             | no               |
+| kimchi memory            | proof-systems | memory   | yes              | yes              | yes             | no               |
+| kimchi memory flamegraph | proof-systems | memory   | no               | yes              | no              | no               |
+| snarky time              | mina/snarky*  | time     | yes              | yes              | yes             | yes              |
+| snarky memory            | mina/snarky*  | memory   | yes              | yes              | yes             | no               |
+| pickles time             | mina/pickles* | time     | yes              | yes              | yes             | yes              |
+| pickles memory           | mina/pickles* | memory   | yes              | yes              | yes             | no               |
+
+\* these have to be in Mina now because pickles is in Mina, and while snarky has
+its own repository, using it requires a few things still in Mina, so that we
+might want to consider some changes to make these benchmarks easier and probably
+cheaper.
+
+### Kimchi time
+
+One of the simplest measuring running time, can be a specific case or allow more
+flexibility by introducing parameters like for example different circuit sizes.
+We already have most of the infrastructure for this with
+[criterion](https://crates.io/crates/criterion). We would ideally have a
+dedicated computer. However, considering that there aren't that many PRs in
+proof-systems, the requirements shouldn't be that high.
+
+### Kimchi time flamegraph
+
+This can be considered a variant of the [previous](#kimchi-time). Some more
+research is required but this information ideally can be generated from the
+already existing benchmarks with little to no changes.  This may not be as
+interesting to have in CI, but can be useful to run locally for those working in
+optimizations.
+
+### Kimchi instructions
+
+This is basically counting the CPU instructions that takes to run some code,
+could be use as an alternative for CI given that it won't be affected by noise
+and can work in any computer. It would also be faster by running the code only
+once. The implementation will require a bit more research but one option is the
```

Comment:
Same with this one re research

Reply:
I think we can settle with `iai`, we already have it as dependency, it could require some more work that just running the time benchmark but should be a mere implementation detail.